chainimport/headers_import.go

```diff
@@ -0,0 +1,1816 @@
+	}
+
+	// Extract import height range.
+	importStartHeight := sourceMetadata.StartHeight
+	importEndHeight := sourceMetadata.EndHeight
+
+	// If import wants to start after height 1, we'd have a gap.
+	if importStartHeight > 1 {
+		return fmt.Errorf("target stores contain only genesis block "+
+			"(height 0) but import data starts at height %d, "+
+			"creating a gap", importStartHeight)
+	}
+
+	// If import includes genesis block (starts at 0), verify it matches.
+	if importStartHeight == 0 {
+		err := s.verifyHeadersAtTargetHeight(importStartHeight)
+		if err != nil {
+			return fmt.Errorf("genesis header mismatch: %v", err)
+		}
+		log.Infof("Genesis headers verified, import data will extend " +
+			"chain from genesis")
+	} else {
+		// Import starts at height 1, which connects to genesis.
+		// Validate that the block header at height 1 from the import
+		// source connects with the previous header in the target block
+		// store.
+		if err := s.validateHeaderConnection(
+			importStartHeight, blockTipHeight, sourceMetadata,
+		); err != nil {
+			return fmt.Errorf("failed to validate header "+
+				"connection: %v", err)
+		}
+
+		log.Info("Target stores contain only genesis block, import " +
+			"data will extend chain from height 1")
+	}
+
+	log.Infof("Chain continuity validation successful: import data "+
+		"(%d-%d) connects properly with target chain",
+		importStartHeight, importEndHeight)
+
+	return nil
+}
+
+// validateHeaderConnection verifies that a block header from the import source
+// properly connects with the previous header in the target block store.
+func (s *HeadersImport) validateHeaderConnection(targetStartHeight,
+	prevTargetBlockHeight uint32,
+	importHeaderMetadata *HeaderMetadata) error {
+
+	// Get the previous block header from target store.
+	prevBlkHdr, err := s.options.TargetBlockHeaderStore.FetchHeaderByHeight(
+		prevTargetBlockHeight,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get block header from "+
+			"target at height %d: %w", prevTargetBlockHeight, err)
+	}
+
+	// Convert target height to the equivalent index for import sources.
+	importSourceIndex := targetHeightToImportSourceIndex(
+		targetStartHeight, importHeaderMetadata.StartHeight,
+	)
+
+	// Get block header at that index from the import source.
+	currHeader, err := s.BlockHeadersImportSource.GetHeader(
+		importSourceIndex,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get block header from "+
+			"import source at height %d: %w",
+			targetStartHeight, err)
+	}
+
+	// Type assert the header retrieved from import source to be block
+	// header.
+	currBlkHeader, ok := currHeader.(*BlockHeader)
+	if !ok {
+		return fmt.Errorf("expected BlockHeader type, got %T",
+			currHeader)
+	}
+
+	// Ensure the current header's previous block hash matches the
+	// hash of the previously fetched block header to maintain chain
+	// integrity.
+	prevHash := prevBlkHdr.BlockHash()
+	if !currBlkHeader.PrevBlock.IsEqual(&prevHash) {
```

Comment:
Shouldn't this be validated as part of the validator, which as already run at this point?